tests/test_state.py

```diff
@@ -196,4 +200,14 @@ def testContextSerialization(self):
         self.assertEqual( new_state.context['step'], 20)
         new_new_state = pickle.loads(new_new_file)
         self.assertEqual( new_new_state.context['step'], 30)
-        
+
+
+    def _test_state_gen_helper(self, name, msg):
+        self.assertEqual(name, 'statename')
+        self.assertEqual(msg, 'statemsg')
+        raise _CallbackExecuted
```

Comment:
Since the assertMethodIsCalled in the link isn't stdlib, I'd prefer not to add it to our codebase when simpler solutions work. To me an exception seemed cleaner than adding some kind of flag variable to the class just for the purpose of this particular test. the fact that the exception stops event handling processing isn't a big issue here since all we care about is if the helper executed, and don't care about anything else, so i'm going to merge as is.